virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -106,9 +107,26 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
     const rightMostPosition = innerWidth - dtWindowWidth;
     const bottomMostPosition = innerHeight - dtWindowHeight;
 
-    const { left: lastTopLeft, top: lastTopRight } = this.lastWindowPosition;
-    let nextLeft = lastTopLeft + WindowManagerService.NEW_WINDOW_POSITION_INCREMENT;
-    let nextTop = lastTopRight + WindowManagerService.NEW_WINDOW_POSITION_INCREMENT;
+    const pluginId = plugin.getIdentifier();
+    if (this.windowPositionsMap.has(pluginId))//If a plugin has previously saved window position & size data
+    { return {
+        ...this.windowPositionsMap.get(pluginId),
+      } as WindowPosition
+    }
+
+    let valueArray = Array.from(this.runningPluginMap.values());
```

Comment:
Why are you cascading indiscriminately? If you are trying to mimic other desktops, they do not do this. Rather, they group instances of each program. So, on Windows for example, it will remember that the last instance of Chrome was at offset 2,55 and will cascade new instances of Chrome from there. However, it will have no bearing on notepad, which may begin cascading from 77,6 if that were its last position.

Reply:
Because that's how Zowe had it now with its indescriminant last window position cascade. It didn't matter what category of applications there were. But I can probably make it so that it does though. Will just require a bit of work, I'll get back to you.